### PR TITLE
[FIX] extended compose file with build context not working

### PR DESCRIPTION
**Pull request by vansari**
Hi @muayyad-alsadi and all other contributers,

With this PR I want to fix the issue with the extended Build-Context which is not working at this time.

I created a sample folder with following structure:

``` 
extends_w_file_subdir
 |--> docker-compose.yml
 |--> sub
   |--> docker-compose.yml
   |--> docker
     |--> example
       |--> Dockerfile
```

If I tried the cmd `./podman_compose.py -f tests/extends_w_file_subdir/docker-compose.yml up -d` I received the message that the docker/example/Dockerfile can not be found.

The goal is to prefix the context path in the extended File with path of the extending File.
I have also wrote a little pytest to check that the container was build successful.

Please can you check if this can solve some reported issues?
Thanks a lot!

PS: Can solve the Issue [#130](https://github.com/containers/podman-compose/issues/130)

### CODE REVIEW

**Comment by filipef9** [comment]
When this fix will be released?

File: podman_compose.py [hunk 1]
```diff
@@ -928,7 +928,15 @@ def volume_ls(self, proj=None):
         volumes = output.splitlines()
         return volumes
 
-def normalize_service(service):
+def normalize_service(service, sub_dir=''):
+    if sub_dir and 'build' in service and 'context' in service['build']:
+        context = service['build']['context']
+        if re.match('^(((\./)?(\w+)+(/\w+)+?$)|(\.$))', context):
```
Comment (muayyad-alsadi) [inline podman_compose.py:934]: I typically compile my regex once, not per-function call.

File: podman_compose.py [hunk 2]
```diff
@@ -928,7 +928,15 @@ def volume_ls(self, proj=None):
...
+            if context.startswith('.'):
+                context = context.replace('.', sub_dir)
+            else:
+                context = sub_dir + os.sep + context
```
Comment (muayyad-alsadi) [inline podman_compose.py:938]: I typically use os.path.join, is their a reason for noting using it?

Comment (muayyad-alsadi) [inline podman_compose.py:938]: I typically use os.path.join, is their a reason for noting using it?

File: podman_compose.py [hunk 3]
```diff
@@ -1008,9 +1016,12 @@ def resolve_extends(services, service_names, environ):
                 content = yaml.safe_load(f) or {}
             if "services" in content:
                 content = content["services"]
+            subdirectory = filename.rsplit(os.sep, 1)[0]
```
Comment (muayyad-alsadi) [inline podman_compose.py:1019]: should this be replace with os.path.basename or os.path.dirname?

File: podman_compose.py [hunk 4]
```diff
@@ -928,7 +928,15 @@ def volume_ls(self, proj=None):
         volumes = output.splitlines()
         return volumes
 
-def normalize_service(service):
+def normalize_service(service, sub_dir=''):
+    if sub_dir and 'build' in service and 'context' in service['build']:
```
Comment (muayyad-alsadi) [inline podman_compose.py:932]: does not seem to handle the case were context is passed as string build like this

```yaml
services:
    web1:
      build: ./context
```

File: podman_compose.py (hunk 1 above)
Comment (muayyad-alsadi) [reply podman_compose.py:934]: I believe the regex is not needed at all, it should be just `.startswith('./')`

Comment (vansari) [reply podman_compose.py:934]: you can hit the path to the context without a './'

File: podman_compose.py (hunk 4 above)
Comment (vansari) [reply podman_compose.py:932]: I will check it

**Comment by vansari** [comment]
@muayyad-alsadi I will create a new PR today with the "fixes" and a newer test for it

**Comment by muayyad-alsadi** [comment]
Please wait. I'm pushing my changes.

**Comment by muayyad-alsadi** [comment]
@vansari please test my last push and I really appreciate your contribution.

**Comment by vansari** [comment]
@muayyad-alsadi Nice work, I have updated my tests in another PR :)

**Pull request merged**